fully-hyperbolic-nn/gcn/grid_search.py

```
@@ -0,0 +1,335 @@
+        'task': args.task,
+        'dataset': args.dataset,
+    }
+    
+    param_grid = ParameterGrid(TUNING_HYPERPARAMETERS)
+
+    for hyperparams in param_grid:
+        combination = {
+            **default_args,
+            **FIXED_PARAMETERS, 
+            **setup,             
+            **hyperparams       
+        }
+        all_combinations.append(combination)
+    
+    return all_combinations
+
+
+def get_setup_name(combination):
+    """
+    Generate a readable name for a setup
+    """
+    return f"{combination['model']}_{combination['optimizer']}_{combination['manifold']}_{combination['dataset']}"
+
+
+def create_results_folder():
+    """
+    Create a folder for this grid search run
+    """
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    results_dir = Path("grid_search_results") / f"run_{timestamp}"
+    results_dir.mkdir(parents=True, exist_ok=True)
+    return results_dir, timestamp
+
+
+def reset_logging():
+    """Reset logging handlers to ensure each experiment gets fresh logging"""
+    root_logger = logging.getLogger()
+    for handler in root_logger.handlers[:]:
+        handler.close()
+        root_logger.removeHandler(handler)
+    logging.shutdown()
+    importlib.reload(logging)
+
+
+def extract_metrics_from_saved_files(save_dir):
+    """
+    Extract validation and test metrics from saved log.txt
+    """
+    try:
+        log_path = Path(save_dir) / 'log.txt'
+        if not log_path.exists():
+            return None, None
+        
+        with open(log_path, 'r') as f:
+            lines = f.readlines()
+        
+        val_acc = None
+        test_acc = None
+        val_loss = None
+        test_loss = None
+        
+        for line in reversed(lines):
+            if 'Val set results:' in line:
+                parts = line.split()
+                for i, part in enumerate(parts):
+                    if part == 'val_acc:' and i + 1 < len(parts):
+                        try:
```

why do you intercept errors everywhere? i think it's better if the script here throws an error and we realize what's wrong rather than silencing the error and moving on? or do i misunderstand the reason